perses/distributed/relative_setup.py

```diff
@@ -307,135 +314,350 @@ class NonequilibriumSwitchingFEP(object):
     }
 
     def __init__(self, topology_proposal, pos_old, new_positions, use_dispersion_correction=False,
-                 forward_functions=None, ncmc_nsteps=100, concurrency=4, platform_name="OpenCL",
-                 temperature=300.0 * unit.kelvin):
+                 forward_functions=None, ncmc_nsteps=100, nsteps_per_iteration=1, concurrency=4, platform_name="OpenCL",
+                 temperature=300.0 * unit.kelvin, trajectory_directory=None, trajectory_prefix=None):
 
+        #construct the hybrid topology factory object
         self._factory = HybridTopologyFactory(topology_proposal, pos_old, new_positions, use_dispersion_correction=use_dispersion_correction)
+
+        #use default functions if none specified
         if forward_functions == None:
             self._forward_functions = self.default_forward_functions
         else:
             self._forward_functions = forward_functions
+
+        #reverse functions to get a symmetric protocol
         self._reverse_functions = {param : param_formula.replace("lambda", "(1-lambda)") for param, param_formula in self._forward_functions.items()}
 
+        #set up some class attributes
         self._hybrid_system = self._factory.hybrid_system
         self._initial_hybrid_positions = self._factory.hybrid_positions
         self._concurrency = concurrency
-
         self._ncmc_nsteps = ncmc_nsteps
+        self._nsteps_per_iteration = nsteps_per_iteration
+        self._trajectory_prefix = trajectory_prefix
+        self._trajectory_directory = trajectory_directory
+
+        #initialize lists for results
+        self._forward_nonequilibrium_trajectories = []
+        self._reverse_nonequilibrium_trajectories = []
+        self._forward_nonequilibrium_cumulative_works = []
+        self._reverse_nonequilibrium_cumulative_works = []
+        self._forward_nonequilibrium_results = []
+        self._reverse_nonequilibrium_results = []
+        self._forward_total_work = []
+        self._reverse_total_work = []
+
+        #Set the number of times that the nonequilbrium move will have to be run in order to complete a protocol:
+        if self._ncmc_nsteps % self._nsteps_per_iteration != 0:
+            logging.warning("The number of ncmc steps is not divisible by the number of steps per iteration. You may not have a full protocol.")
+        self._n_iterations_per_call = self._ncmc_nsteps // self._nsteps_per_iteration
+
+        #create the thermodynamic state
+        lambda_zero_alchemical_state = alchemy.AlchemicalState.from_system(self._hybrid_system)
+        lambda_one_alchemical_state = copy.deepcopy(lambda_zero_alchemical_state)
+
+        #ensure their states are set appropriately
+        lambda_zero_alchemical_state = self._set_all_parameters_of_state(lambda_zero_alchemical_state, 0.0)
```

Comment:
You should call `lambda_zero_alchemical_state.set_alchemical_parameters(0.0)`. I think you'll get an error later if you set to `0.0` a lambda parameter that is not defined in the `System` (like `lambda_bonds` in most cases). Parameters that are not defined in the system should be `None` (the rationale for the error is to make it clear that you're not modifying anything even if you change the state's parameter).

Reply:
ok, will do. I have all of those parameters, but to be safe I'll use the proper API.